### PR TITLE
[BrowserKit] Rename Client to Browser

### PR DESCRIPTION
**Pull request by fabpot**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no     
| Deprecations? | yes
| Tests pass?   | yes    
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

`Client` is very generic and used in 3 places: BrowserKit, HttpKernel, and FramewrokBundle. Each Client extends another one. So, to make things clearer, I'd like to rename Client to Browser like this:

Symfony\Component\BrowerKit\Client -> AbstractBrowser
Symfony\Component\HttpKernel\Client -> HttpKernelBrowser
Symfony\Bundle\FrameworkBundle\Client -> KernelBrowser

The next PR will introduce an `HttpBrowser` based on the new HttpClient component :)

### CODE REVIEW

**Review by nicolas-grekas — APPROVED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Test/WebTestCase.php [hunk 1]
```diff
@@ -22,12 +22,12 @@
...
-     * Creates a Client.
+     * Creates a KernelBrowser.
      *
      * @param array $options An array of options to pass to the createKernel method
      * @param array $server  An array of server parameters
      *
-     * @return Client A Client instance
+     * @return KernelBrowser A KernelBrowser instance
      */
     protected static function createClient(array $options = [], array $server = [])
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/Test/WebTestCase.php]: should we rename that method too ?

**Comment by stof** [comment]
This is a BC break, because the BC class `Symfony\Bundle\FrameworkBundle\Client` does not extend the BC class `Symfony\Component\BrowerKit\Client`

**Comment by stof** [comment]
I think you will need to use class aliases here instead., to preserve the class hierarchy.

File: src/Symfony/Bundle/FrameworkBundle/Test/WebTestCase.php (hunk 1 above)
Comment (fabpot) [reply src/Symfony/Bundle/FrameworkBundle/Test/WebTestCase.php]: I thought about renaming methods but I think that's too much. What others think?

Comment (HeahDude) [reply src/Symfony/Bundle/FrameworkBundle/Test/WebTestCase.php]: With a hard copy pasting you could also add return type hints on methods.

**Comment by Devristo** [comment]
Yes, it breaks at least `https://github.com/FriendsOfBehat/SymfonyExtension` and `https://github.com/minkphp/MinkBrowserKitDriver`.

**Pull request merged**